chapter-06/inprogress/bookworm.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"sort"
+)
+
+// A Bookworm contains the list of books on a bookworm's shelf.
+type Bookworm struct {
+	Name  string `json:"name"`
+	Books []Book `json:"books"`
+}
+
+// Book describes a book on a bookworm's shelf.
+type Book struct {
+	Authors string `json:"authors"`
+	Title   string `json:"title"`
+}
+
+// loadBookworms reads the file and returns the list of bookworms, and their beloved books, found therein.
+func loadBookworms(filePath string) ([]Bookworm, error) {
+	f, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	contents, err := io.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+
+	bookworms := make([]Bookworm, 0)
+	err = json.Unmarshal(contents, &bookworms)
```

Comment:
Use Unmarshal or Decode. I don't know which is best, but I'd like both to appear in the book.

Unmarshal works great when the data is already in the memory, whereas Decode works great on a reader. I think I'll personnaly favour Decode over Unmarshal in most cases.

If we use Unmarshal, we shouldn't use os.Open + io.ReadAll, but instead os.ReadFile()

Reply:
Fixed using `Decode`.